buildrunner/validation/config.py

```diff
@@ -132,6 +138,155 @@ def validate_push(
+                is_multi_platform = True
+
+            return src_image, is_multi_platform
+
+        def get_destination_images(step: Step) -> Optional[List[str]]:
+            """
+            Get the destination images from step.push or step.commit
+
+            Args:
+                step (Step): Build step
+
+            Returns:
+                Optional[List[str]]: List of destination images
+            """
+
+            def get_images(
+                pushcommmit: Union[
+                    StepPushCommitDict, str, List[Union[str, StepPushCommitDict]]
+                ],
+            ) -> List[str]:
+                images = []
+                if isinstance(pushcommmit, StepPushCommitDict):
+                    if pushcommmit.tags:
+                        for tag in pushcommmit.tags:
+                            images.append(f"{pushcommmit.repository}:{tag}")
+                    else:
+                        images.append(f"{pushcommmit.repository}:latest")
+                elif isinstance(pushcommmit, str):
+                    if ":" not in pushcommmit:
+                        images.append(f"{pushcommmit}:latest")
+                    else:
+                        images.append(pushcommmit)
+                elif isinstance(pushcommmit, list):
+                    for item in pushcommmit:
+                        if isinstance(item, str):
+                            images.append(item)
+                        elif isinstance(item, StepPushCommitDict):
+                            if item.tags:
+                                for tag in item.tags:
+                                    images.append(f"{item.repository}:{tag}")
+                            else:
+                                images.append(f"{item.repository}:latest")
+                        else:
+                            raise ValueError(
+                                f"Unknown type for step.push: {type(step.push)}"
+                            )
+                return images
+
+            images = []
+            images.extend(get_images(step.push))
+            images.extend(get_images(step.commit))
+            return images
+
+        def validate_multiplatform_are_not_retagged():
+            """
+            Validate multi-platform are not re-tagged
+
+            Args:
+                mp_push_tags (Set[str]): Set of all tags used in multi-platform build steps
+
+            Raises:
+                ValueError | pydantic.ValidationError: If the config file is invalid
+            """
+            step_images = {}
+
+            # Iterate through each step and get information about the images
+            for step_name, step in vals.items():
+                src_image, is_multi_platform = get_source_image(step)
+                dst_images = get_destination_images(step)
+                step_images[step_name] = StepImagesInfo(
+                    source_image=src_image,
+                    dest_images=dst_images,
```

Comment:
Why not just make these args match the parameter names for simplicity? (i.e. `source_image` and `dest_images`)

Reply:
Fixed